A47-xds-federation.md

```diff
@@ -0,0 +1,682 @@
+//   field turns it into an "xdstp:" URI.
+// - A gRPC server is created and the
+//   "server_listener_resource_name_template" field is an "xdstp:" URI.
+//
+// In any of those cases, it is an error if the specified authority is
+// not present in this map.
+"authorities": {
+  // Entries are keyed by authority name.
+  // Note: If a new-style resource name has no authority, we will use
+  // the empty string here as the key.
+  "<authority_name>": {
+    // A template for the name of the Listener resource to subscribe
+    // to for a gRPC client channel.  Used only when the channel is
+    // created using an "xds:" URI with this authority name.
+    //
+    // The token "%s", if present in this string, will be replaced
+    // with %-encoded service authority (i.e., the path part of the target
+    // URI used to create the gRPC channel).
+    //
+    // Must start with "xdstp://<authority_name>/".  If it does not,
+    // that is considered a bootstrap file parsing error.
+    //
+    // If not present in the bootstrap file, defaults to
+    // "xdstp://<authority_name>/envoy.config.listener.v3.Listener/%s".
+    "client_listener_resource_name_template": <string>,
+
+    // Ordered list of xDS servers to contact for this authority.
+    // Format is exactly the same as the top level "xds_servers" field.
+    //
+    // If the same server is listed in multiple authorities, the
+    // entries will be de-duped (i.e., resources for both authorities
+    // will be fetched on the same ADS stream).
+    //
+    // If not specified, the top-level server list is used.
+    "xds_servers": [ ... ]
+  }
+}
+```
+
+The new `client_default_listener_resource_name_template` field will have
+the following format:
+
+```
+// A template for the name of the Listener resource to subscribe to
+// for a gRPC client channel.  Used only when the channel is created
+// with an "xds:" URI with no authority.
+//
+// If starts with "xdstp:", will be interpreted as a new-style name,
+// in which case the authority of the URI will be used to select the
+// relevant configuration in the "authorities" map.
+//
+// The token "%s", if present in this string, will be replaced with
+// the service authority (i.e., the path part of the target URI
+// used to create the gRPC channel).  If the template starts with
+// "xdstp:", the replaced string will be %-encoded.
+//
+// Defaults to "%s".
+"client_default_listener_resource_name_template": <string>,
+```
+
+There will be two changes to the existing
+`server_listener_resource_name_template` described in [gRFC A36][A36]:
+- If the template starts with `xdstp:`, it will be interpreted as a
+  new-style name, in which case the authority of the URI will be used to
+  select the relevant configuration in the `authorities` map.
+- When replacing the `%s` token, if the template starts with `xdstp:`,
+  the replacement string will be %-encoded in the resulting resource
```

Comment:
The doc uses both %-encoding or percent-encoding. Will be good to stick to just one term.

Reply:
Changed to use "percent-encoding" everywhere.